pkg/operator/ceph/object/realm/controller_test.go

```diff
@@ -0,0 +1,250 @@
+	r := &ReconcileObjectRealm{client: cl, scheme: s, context: c, clusterInfo: cInfo}
+
+	// Mock request to simulate Reconcile() being called on an event for a
+	// watched resource .
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	logger.Info("STARTING PHASE 1")
+	res, err := r.Reconcile(req)
+	assert.NoError(t, err)
+	assert.True(t, res.Requeue)
+	logger.Info("PHASE 1 DONE")
+
+	//
+	// TEST 2:
+	//
+	// FAILURE we have a cluster but it's not ready
+	//
+	cephCluster = &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespace,
+			Namespace: namespace,
+		},
+		Status: cephv1.ClusterStatus{
+			Phase: "",
+			CephStatus: &cephv1.CephStatus{
+				Health: "",
+			},
+		},
+	}
+	object = append(object, cephCluster)
+	// Create a fake client to mock API calls.
+	cl = fake.NewFakeClientWithScheme(s, object...)
+	// Create a ReconcileObjectRealm object with the scheme and fake client.
+	r = &ReconcileObjectRealm{client: cl, scheme: s, context: c}
+	logger.Info("STARTING PHASE 2")
+	res, err = r.Reconcile(req)
+	assert.NoError(t, err)
+	assert.True(t, res.Requeue)
+	logger.Info("PHASE 2 DONE")
+	logger.Info("TEST CEPH-OBJECT-REALM-CONTROLLER DONE")
+}
+
+func TestPullCephRealm(t *testing.T) {
+	logger.Info("STARTING TEST PULL-CEPH-REALM")
+	// Set DEBUG logging
+	capnslog.SetGlobalLogLevel(capnslog.DEBUG)
+
+	r, objectRealm := getObjectRealmAndReconcileObjectRealm(t)
+
+	secrets := map[string][]byte{
+		"access-key": []byte("akey"),
+		"secret-key": []byte("skey"),
+	}
+
+	secretName := objectRealm.Name + "-keys"
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: objectRealm.Namespace,
+		},
+		Data: secrets,
+		Type: k8sutil.RookType,
+	}
+
+	_, err := r.context.Clientset.CoreV1().Secrets(objectRealm.Namespace).Create(secret)
+	assert.NoError(t, err)
+
+	objectRealm.Spec.Pull.Endpoint = "http://10.2.1.164:80"
+	res, err := r.pullCephRealm(objectRealm)
+	assert.NoError(t, err)
+	assert.False(t, res.Requeue)
+	logger.Info("TEST PULL-CEPH-REALM DONE")
+}
+
+func TestCreateRealmKeys(t *testing.T) {
+	logger.Info("STARTING TEST CREATE-REALM-KEYS")
+	// Set DEBUG logging
+	capnslog.SetGlobalLogLevel(capnslog.DEBUG)
+
+	r, objectRealm := getObjectRealmAndReconcileObjectRealm(t)
+
+	res, err := r.createRealmKeys(objectRealm)
+	assert.NoError(t, err)
+	assert.False(t, res.Requeue)
+	logger.Info("TEST CREATE-REALM-KEYS DONE")
+}
+
+func TestCreateCephRealm(t *testing.T) {
+	logger.Info("STARTING TEST CREATE-CEPH-REALM")
+	// Set DEBUG logging
+	capnslog.SetGlobalLogLevel(capnslog.DEBUG)
+
+	r, objectRealm := getObjectRealmAndReconcileObjectRealm(t)
+
+	res, err := r.createCephRealm(objectRealm)
+	assert.NoError(t, err)
+	assert.False(t, res.Requeue)
+	logger.Info("TEST CREATE-CEPH-REALM DONE")
+}
+
+func getObjectRealmAndReconcileObjectRealm(t *testing.T) (*ReconcileObjectRealm, *cephv1.CephObjectRealm) {
```

Comment:
Can you call this helper from `TestCephObjectRealmController()` as well? Looks like duplicate code.